packages/flutter_tools/lib/src/xcode_project.dart

```diff
@@ -327,6 +330,24 @@ class IosProject extends XcodeBasedProject {
     return null;
   }
 
+  BuildInfo _generateBuildInfoFrom(XcodeProjectBuildContext context) {
+    late BuildMode mode;
+    if (context.configuration == null) {
+      mode = BuildMode.release;
+    } else {
+      try {
+        mode = BuildMode.values.byName(context.configuration!);
+      } on ArgumentError {
+        mode = BuildMode.release;
+      }
+    }
+    return BuildInfo(
+      mode,
+      context.scheme,
+      treeShakeIcons: true,
+    );
+  }
```

Comment:
So I worry that if we add this, it may get used later on for other functionality, not realizing that the `BuildInfo` is not complete. 

I think it would be better to pass the `BuildInfo` here 
https://github.com/flutter/flutter/blob/8e431e106e2b9cf1e2570f4cc13f9344c8699534/packages/flutter_tools/lib/src/commands/analyze.dart#L290-L296

Perhaps something like
```dart
      BuildMode mode;
      try {
        mode = BuildMode.values.byName(configuration!);
      } on ArgumentError {
        mode = BuildMode.release;
      }
      final BuildInfo buildInfo = await getBuildInfo(forcedBuildMode: mode);
      await IOSAnalyze(
        project: FlutterProject.fromDirectory(_fileSystem.directory(directoryPath)),
        option: option,
        configuration: configuration,
        target: target,
        logger: _logger,
        buildInfo: buildInfo,
      ).analyze();
```